x/feegrant/genesis.go

```diff
@@ -6,46 +6,29 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/feegrant/types"
 )
 
-// GenesisState contains a set of fee allowances, persisted from the store
-type GenesisState []types.FeeAllowanceGrant
-
-// ValidateBasic ensures all grants in the genesis state are valid
-func (g GenesisState) ValidateBasic() error {
-	for _, f := range g {
-		grant, err := f.GetFeeGrant()
-		if err != nil {
-			return err
-		}
-		err = grant.ValidateBasic()
-		if err != nil {
-			return err
-		}
-	}
-	return nil
-}
```

Comment:
Why are we removing this validation here?

Reply:
`type GenesisState []types.FeeAllowanceGrant` This declaration is no longer required, I think we are using `GenesisState` from proto definitions that's why these lines removed. I guess this (https://github.com/cosmos/cosmos-sdk/blob/marie/9115-feegrant-updates/x/feegrant/types/genesis.go#L15-L27) validation is being used for genesis validation.

Reply:
exactly